e2etest/connectivity_two_cluster_test.go

```diff
@@ -0,0 +1,155 @@
+				LabelSelector: metav1.FormatLabelSelector(helloDeployment.Spec.Selector),
+			})
+			Expect(err).ToNot(HaveOccurred())
+			if len(pods.Items) == 0 {
+				return ""
+			}
+			return pods.Items[0].Status.PodIP
+		}, 30).ShouldNot(BeEmpty())
+		pods1, err := cluster1.k8s.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: metav1.FormatLabelSelector(helloDeployment.Spec.Selector),
+		})
+		Expect(err).ToNot(HaveOccurred())
+		pods2, err := cluster2.k8s.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: metav1.FormatLabelSelector(helloDeployment.Spec.Selector),
+		})
+		Expect(err).ToNot(HaveOccurred())
+		for _, pod := range slices.Concat(pods1.Items, pods2.Items) {
+			podIPs = append(podIPs, pod.Status.PodIP)
+		}
+
+		exportService(ctx, cluster2, cluster1, namespace, svc.Name)
+		exportService(ctx, cluster1, cluster1, namespace, svc.Name)
+
+		Eventually(func() []string {
+			svcImport, err := cluster1.mcs.MulticlusterV1alpha1().ServiceImports(namespace).Get(ctx, helloServiceImport.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return svcImport.Spec.IPs
+		}).ShouldNot(BeEmpty())
+		serviceImport, err = cluster1.mcs.MulticlusterV1alpha1().ServiceImports(namespace).Get(ctx, helloServiceImport.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		reqPod, err = cluster1.k8s.CoreV1().Pods(namespace).Get(ctx, requestPod.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		By("Created all in " + namespace)
+	})
+	AfterEach(func() {
+		if *noTearDown {
+			By(fmt.Sprintf("Skipping teardown. Test namespace %q", namespace))
+			By(fmt.Sprintf("Cluster 1: kubectl --kubeconfig %q -n %q", *kubeconfig1, namespace))
+			By(fmt.Sprintf("Cluster 2: kubectl --kubeconfig %q -n %q", *kubeconfig2, namespace))
+			return
+		}
+		Expect(cluster1.k8s.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})).To(Succeed())
+		Expect(cluster2.k8s.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})).To(Succeed())
+	})
+	Specify("UDP connects across clusters using the VIP", func() {
+		command := []string{"sh", "-c", fmt.Sprintf("echo hi | nc -uw1 %s 42", serviceImport.Spec.IPs[0])}
+		checkPodReachable(command)
+	})
+	Specify("TCP connects across clusters using the VIP", func() {
+		command := []string{"sh", "-c", fmt.Sprintf("echo hi | nc %s 42", serviceImport.Spec.IPs[0])}
+		checkPodReachable(command)
+	})
+	Specify("UDP connects across clusters using the DNS name", func() {
+		command := []string{"sh", "-c", fmt.Sprintf("echo hi | nc -uw1 %s.%s.svc.clusterset.local 42", serviceImport.Name, serviceImport.Namespace)}
+		checkPodReachable(command)
+	})
+	Specify("TCP connects across clusters using the DNS name", func() {
+		command := []string{"sh", "-c", fmt.Sprintf("echo hi | nc %s.%s.svc.clusterset.local 42", serviceImport.Name, serviceImport.Namespace)}
+		checkPodReachable(command)
+	})
+})
```

Comment:
There is a lot of code duplicated with _connectivity_test.go_.  It looks like much of it could be reused. The vars above on L37-40 could be moved to a struct and the common code moved to methods on the struct.

Reply:
I refactored this to slot these new tests into the existing connectivity test definition.

Reply:
👍